src/mscorlib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -430,19 +443,14 @@ public override void BeginCatchBlock(Type exceptionType)
 
                 // if this is a catch block the exception will be pushed on the stack and we need to update the stack info
                 UpdateStackSize(OpCodes.Nop, 1);
-            }
 
-            current.MarkCatchAddr(ILOffset, exceptionType);
+                current.MarkCatchAddr(ILOffset, exceptionType);
 
 
-            // this is relying on too much implementation details of the base and so it's highly breaking
-            // Need to have a more integreted story for exceptions
-            current.m_filterAddr[current.m_currentCatch - 1] = GetTokenFor(rtType);
-        }
-
-        public override void BeginFaultBlock()
-        {
-            throw new NotSupportedException(Environment.GetResourceString("InvalidOperation_NotAllowedInDynamicMethod"));
+                // this is relying on too much implementation details of the base and so it's highly breaking
+                // Need to have a more integreted story for exceptions
+                current.m_filterAddr[current.m_currentCatch - 1] = GetTokenFor(rtType);
+            }
         }
 
         public override void BeginFinallyBlock()
```

Comment:
I was wondering if I should do the opposite, to maintain consistency or option 3; clear out all overrides that just pass back to the base. I think I'll do that last option, but in a separate commit in case someone points out a problem with that.